jb/child_allowance.py

```diff
@@ -0,0 +1,236 @@
+# Roadmap:
+# Comparison to other potential policies
+# Replicate plotly code from the child allowance for JB
+# Move to second simulation
+
+# Preamble and read data
+import microdf as mdf
+import pandas as pd
+import numpy as np
+import us
+
+# Read in census data and specify columns for use
+raw = pd.read_csv(
+    "https://github.com/UBICenter/child-allowance/blob/master/jb/data/cps_00003.csv.gz",
+    usecols=[
+        "YEAR",
+        "MONTH",
+        "STATEFIP",
+        "AGE",
+        "SEX",
+        "RACE",
+        "HISPAN",
+        "SPMWT",
+        "SPMFTOTVAL",
+        "SPMTOTRES",
+        "SPMCHXPNS",
+        "SPMTHRESH",
+        "SPMFAMUNIT",
+        "ASECWT",
+    ],
+)
+
+# Create a copy of the raw dataset and make column names non-capitalized
+# for readability
+person = raw.copy(deep=True)
+person = person[~person.ASECWT.isnull()]
+person.columns = person.columns.str.lower()
+
+# Asec weights are year-person units, and we average over 3 years,
+# so divide by the 3 years to give per year weights.
+person.asecwt /= 3
+
+# Define child age identifiers
+person["child_6"] = person.age < 6
+person["infant"] = person.age < 1
+person["toddler"] = person.age.between(1, 2)
+person["preschool"] = person.age.between(3, 5)
+person["person"] = 1
+
+# Redefine race categories
+person.race.mask(
```

Comment:
Could you make this a new column instead, e.g. `person.race_group`? Often safer to avoid overloading original column names with new concepts. Same with `hispan` below.